### PR TITLE
Restructuring modeling

### PR DESCRIPTION
**Pull request by ElieBendier**
Category table doesn't exist anymore.
Category is now an element of the video entity.
I have made all the changes needed for our existing features to work with these changes, which include:
- index language
- index category
- index video
- player video
- new video

### CODE REVIEW

File: src/Controller/VideoController.php [hunk 1]
```diff
@@ -32,56 +31,62 @@ public function new(
...
         ]);
     }
 
+    #[Route('/show/{languageSlug}/{videoSlug}', name: 'show')]
+    public function show(
+        string $languageSlug,
+        string $videoSlug,
+        VideoRepository $videoRepository,
+    ): Response {
+        $video = $videoRepository->findOneBy(['slug' => $videoSlug]);
```
Comment (mickaellambert) [inline src/Controller/VideoController.php]: Récupérer par slug ET par langage

File: src/Entity/Video.php [hunk 2]
```diff
@@ -12,11 +12,11 @@
 #[Vich\Uploadable]
 class Video
 {
+    public const CATEGORIES = ['Cours' => 'Cours', 'Tutoriel' => 'Tutoriel', 'LiveCoding' => 'LiveCoding'];
```
Comment (mickaellambert) [inline src/Entity/Video.php]:
```
public const CATEGORIES = [
    'Cours', 
    'Tutoriel', 
    'LiveCoding'
];
```

File: src/Controller/LanguageController.php [hunk 3]
```diff
@@ -17,4 +18,25 @@ public function index(LanguageRepository $languageRepository): Response
...
     }
+
+    #[Route('/{slug}', name: 'category_index')]
+    public function listCategoryByLanguage(
+        string $slug,
+        VideoRepository $videoRepository,
+        LanguageRepository $languageRepository
+    ): Response {
+        $language = $languageRepository->findBySlug($slug);
+        $results = $videoRepository->findByLanguage($language);
```
Comment (mickaellambert) [inline src/Controller/LanguageController.php]: `$categories = $videoRepository->findDistinctCategories($language); `

File: src/Controller/LanguageController.php [hunk 4]
```diff
@@ -17,4 +18,25 @@ public function index(LanguageRepository $languageRepository): Response
...
+
+        $videos = [];
+
+        foreach ($results as $result) {
+            $videos[$result->getCategory()][] = $result;
+        }
+
+        return $this->render('category/index.html.twig', [
+            'videos' => $videos,
```
Comment (mickaellambert) [inline src/Controller/LanguageController.php:34]: 'categories' => $categories

File: templates/category/index.html.twig [hunk 5]
```diff
@@ -10,13 +10,13 @@
 	<div id="index-category">
 		<h1>//CATÉGORIES_</h1>
 		<div id="index-category-cards">
-			{% for category in categories %}
+			{% for key, category in videos %}
```
Comment (mickaellambert) [inline templates/category/index.html.twig]: `for category in categories`

File: templates/category/index.html.twig [hunk 6]
```diff
@@ -10,13 +10,13 @@
...
 				<div class="category-carousel">
-					<h2>{{category.label}}</h2>
-					<section class="splide" id="{{category.label}}-slider" aria-label="Splide Basic HTML Example">
+					<h2>{{key}}</h2>
```
Comment (mickaellambert) [inline templates/category/index.html.twig]: category

**Review by mickaellambert — CHANGES_REQUESTED** [review]

File: src/Controller/VideoController.php (hunk 1 above)
Comment (ElieBendier) [reply src/Controller/VideoController.php]: Fait.

File: src/Entity/Video.php (hunk 2 above)
Comment (ElieBendier) [reply src/Entity/Video.php]: fait.

File: src/Controller/LanguageController.php (hunk 3 above)
Comment (ElieBendier) [reply src/Controller/LanguageController.php]: fait

File: src/Controller/LanguageController.php (hunk 4 above)
Comment (ElieBendier) [reply src/Controller/LanguageController.php:34]: fait.

File: templates/category/index.html.twig (hunk 5 above)
Comment (ElieBendier) [reply templates/category/index.html.twig]: fait

File: templates/category/index.html.twig (hunk 6 above)
Comment (ElieBendier) [reply templates/category/index.html.twig]: fait

**Review by mickaellambert — APPROVED** [review]

**Pull request merged**